src/ray/raylet/scheduling_resources.h

```diff
@@ -8,25 +8,64 @@
 
 #include "ray/raylet/format/node_manager_generated.h"
 
-#define EPSILON 0.00001
-
 namespace ray {
 
 namespace raylet {
 
 const std::string kCPU_ResourceLabel = "CPU";
 
-/// \brief Test if the quantity is within epsilon bounds of 0.
-///
-/// \param quantity: Quantity to check
-/// \return True if  -epsilon <= Quantity <= epsilon, False otherwise.
-bool EqualsZeroEpsilon(double quantity);
+/// \class FractionalResourceQuantity
+/// \brief Converts the resource quantities to an internal representation to
+/// avoid machine precision errors.
+class FractionalResourceQuantity {
+ public:
+  /// \brief Construct a FractionalResourceQuantity representing zero
+  /// resources.
+  FractionalResourceQuantity();
+
+  /// \brief Construct a FractionalResourceQuantity for implicit conversion of
+  /// operators.
+  FractionalResourceQuantity(int resource_quantity);
```

Comment:
Why do we need this constructor? Can we get rid of it?

Reply:
This constructor is only used to implicitly convert integers to `FractionalQuantityResources` when doing logical comparisons such as the below instance
https://github.com/ray-project/ray/pull/4533/files/1592ead002086768b8ec6ea609bd9a625a1dc08d#diff-d19029e3f18a5a52083a2f0bb93da385R291